### ISSUE TITLE
Make replace pattern in `regex.replace` aware of matching groups.

### ISSUE THREAD

**Issue by mximp** [objectionary/eo-strings#30]
Make it possible to refer to matching groups by index like in the example below:
```
[] > app
  stdout > @
    replace
      "a1a A2B B3B"
      "/((([A-Za-z])[0-9])\g{2})/"
      "$2"
```
Output:
```
"a1 A2B B3"
```

For more information see objectionary/eo#865 description and [Java Matcher](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/regex/Matcher.html#replaceAll(java.lang.String))

**Comment by yegor256** [comment]
@levBagryansky transferred

**Comment by levBagryansky** [comment]
> Make it possible to refer to matching groups by index like in the example below:
> 
> ```
> [] > app
>   stdout > @
>     replace
>       "a1a A2B B3B"
>       "/((([A-Za-z])[0-9])\g{2})/"
>       "$2"
> ```

@mximp do you mean 
```
[] > app
  stdout > @
    replaced.
      compile.
        QQ.txt.regex
        "a1a A2B B3B"
      "/((([A-Za-z])[0-9])\g{2})/"
      "$2"
```
?
Is 2 index of group here?
So must `"/((([A-Za-z])[0-9])\g{2})/"` encounters be replaced by the 2nd group of this regex?

**Comment by levBagryansky** [comment]
@mximp It has been implemented in #86 pr. You can close the issue.
